src/renderer/git/log.js

```diff
@@ -1,16 +1,31 @@
-import git from "simple-git/promise";
-import * as Sentry from "@sentry/electron";
-
-const log = async (repository, params) => {
-	const data = await git(repository.path).log(params);
-	try {
-		return data.all;
-	} catch (error) {
-		Sentry.captureException(error);
-		let errorMessage = "Unable to fetch logs.";
-		console.log(errorMessage);
-		Sentry.captureMessage(errorMessage, data);
-	}
+import nodegit, { Repository } from "nodegit";
+
+const log = async path => {
+	const repo = await Repository.open(path);
+	const commit = await repo.getHeadCommit();
+
+	// History returns an event.
+	const history = commit.history(nodegit.Revwalk.SORT.TIME);
+
+	let logs = [];
+
+	// History emits "commit" event for each commit in the branch's history
+	history.on("commit", function(commit) {
+		let objectCommit = {
+			author_email: commit.author().email(),
+			author_name: commit.author().name(),
+			body: commit.body(),
+			date: commit.date(),
+			hash: commit.sha(),
+			message: commit.message()
+		};
+
+		logs.push(objectCommit);
+	});
```

Comment:
I am still wondering about this part. And previously also I was facing the same issue getting more than 500+ items in array with bunch of data can make the app goes really slow.

Reply:
You are right, may be we can do something like pagination? 
May be ask to the function `n` numbers of history items, what do you think?